src/programs/ectrans-benchmark.F90

```diff
@@ -581,16 +585,22 @@ program transform_test
 ztstepmax2 = 0._jprd
 ztstepmin2 = 9999999999999999._jprd
 
-write(nout,'(a)') '======= Start of spectral transforms  ======='
-write(nout,'(" ")')
+if (verbosity >= 1 .and. myproc == 1) then
+  write(nout,'(a)') '======= Start of spectral transforms  ======='
+  write(nout,'(" ")')
+endif
 
 ztloop = timef()
 
 !===================================================================================================
 ! Do spectral transform loop
 !===================================================================================================
 
-do jstep = 1, iters
+gstats_lstats = .false.
+
+do jstep = 1, iters+2
```

Comment:
@marsdeno any ideas why your branch does 2 extra iterations?

Reply:
hope it is fine if I also put some comments :) see the line below, the idea is to turn on gstats in the thrid timestep to exclude the setup time. creating the fft plans can be expensive and is not really something we should include in "per-timestep" timers.

Reply:
That's reasonable. Maybe we should add a print statement to explain this briefly as I was confused for a while why I got 102 iterations when I asked for 100.

Reply:
I think a comment should definitely be added yes.
Also, I think we should refactor the code (followup) to have setup costs in `setuptrans` and not during the time stepping loop, and then this could be reverted.

Reply:
I agree with adding a comment (or maybe a print statement saying "100 perf it + 2 warmup it").  The second part I disagree with. It might be possible for this benchmark to move set up costs into setup, but not in general. Also, it was never the case, because you would need to know beforehand all sizes for which spectral transforms are being called. This is possible for the benchmark here, but not representative for the real world. I think, we could evolve ectrans in the direction of creating handles similar to fft plans, so the users could create the "ectrans" handles during setup if they want (which would be a very nice direction because it would add better encapsulation of many global module variables we have today). But in the end, it is just a fact that the first 1-2, maybe more timesteps will be slower, be it due to memory registration during MPI, be it because of allocation overhead, be it because of FFT plan creation. It is always saver to exclude those.

Reply:
Right "warmup" iterations are always useful. I was not thinking of memory/cache/registration, ... and spoke too soon. I was only considering the extra computations required on first use.

Still having the FFT plan creation should in principle be feasible in setuptrans (not setuptrans0 !) because all runtime options (grid, resolution, domain decomposition) should be known at that point.

Reply:
No, I don't think so. The problem is that FFT plan creation requires knowledge of number of levels, which depends on the configuration with which we call the spectral transform; so this is a property of dir_trans/inv_trans, not of setup_trans (if we call it once with a single level, and then with lvordiv=true, and once with lvordiv=false, three different sets of transforms are generated). Even worse, the graphs that are built on top, using the FFT plans, require the pointers of the intermediate/internal buffer. If the pointers change, graphs are destroyed and recreated, which can happen until we found the largest intermediate buffer. What I was using to avoid this in "production runs" is that I just have an env variable to set the minimum size of the internal buffer. This you only have to figure out once, and then you can set the env variable. Any graph will be created only once, and you don't pay for it many times. But maybe we can postpone this discussion and just add the comment for now. We should probably not strive for the perfect solution in the first step, otherwise we will never merge (and I am sure you agree with this :D )

Reply:
I've added a print statement to clarify the number of iterations.

Reply:
I see. I also never intended this to be a blocker for this PR, just a follow-up discussion. Thanks for your insights @lukasm91 !